web_app/src/app/form1/page.tsx

```diff
@@ -13,53 +12,80 @@ import { Step1Data, Step2Data, Step3Data, Step4Data } from '../interfaces';
 import { generatePDF } from '../components/GeneratePDF';
 import { postForm1, sendEmail } from './actions';
 import dayjs from 'dayjs';
+import { useRouter } from 'next/navigation';
 
 export default function Page() {
 
-    const SUCCESS_MESSAGE = "Your Request for review is sent. Please check your email.";
+    const apiSubmitErrorMsg = "An error occurred while submitting the application. Please try again or contact RoadSafetyBC by calling 1-855-387-7747 and select option 5. You may also attend a driver licensing office to apply for a review.";
+    const pdfEmailSubmitError = "An error occurred while submitting pdf email. Please try again or contact RoadSafetyBC by calling 1-855-387-7747 and select option 5. You may also attend a driver licensing office to apply for a review.";
 
-    const step1Ref = useRef<{ clearData: () => void }>(null);
+    const router = useRouter();
+    const [progress, setProgress] = useState(0);
+
+    const step1Ref = useRef<{ clearData: () => void, validate: () => boolean }>(null);
     const step2Ref = useRef<{ clearData: () => void, validate: () => boolean }>(null);
     const step3Ref = useRef<{ clearData: () => void, validate: () => boolean }>(null);
     const step4Ref = useRef<{ clearData: () => void, validate: () => void }>(null);
 
     const [isExpanded, setIsExpanded] = useState<boolean>(false);
     const [isLoading, setIsLoading] = useState(false);
     const [message, setMessage] = useState('');
-    const [submitError, setSubmitError] = useState<boolean>(false);
 
+    const combineError = (errorMsg: string, newError: string) => {
+        if (errorMsg.startsWith("There are error in"))
+            return errorMsg + ", " + newError;
+        else
+            return "There are error in " + newError;
```

Comment:
Grammer is unusal here. Should this be "There is an error " or "There are errors "?

Reply:
yes, agree will make the change